src/src/com/microsoft/aad/adal/Oauth2.java

```diff
@@ -516,12 +517,12 @@ private AuthenticationResult postMessage(String requestMessage, HashMap<String,
 
                 Logger.v(TAG, "Server error message:" + errMessage);
                 if (response.getResponseException() != null) {
-                    throw response.getResponseException();
+                    throw new IOException(response.getResponseException());
```

Comment:
True, but it unsafe just to convert (in case if some other developer starts to set some other exceptions to this variable), let's make this exception implicitly IOException